### PR TITLE
feat: editor file tree

### PR DESCRIPTION
**Pull request by serkodev**
- [x] Create basic file tree
- [x] Adjust tree view ui (full width row item)
- [x] Sort by file name and put folders on top

Screenshot

### CODE REVIEW

**Comment by SSMM0498** [comment]
👋 Hi, do you planned to add create, rename, move and delete features to that?

**Comment by serkodev** [comment]
I personally believe that these features will be needed in the future. For instance, some tutorials may require operations to create files.
Let's wait for this PR to be reviewed. If the direction is correct, then maybe we can continue to develop. Welcome to cooperate. 💪🏻

File: templates/utils.ts [hunk 1]
```diff
@@ -30,3 +31,33 @@ export function filesToWebContainerFs(
   }
   return tree
 }
+
```
Comment (serkodev) [inline templates/utils.ts:34]: I think the code here (`filesToVirtualFsTree` function) is somewhat repetitive above, but I'm not sure whether it should be merged with the functionality above, or if the whole approach should be redesigned.

File: templates/utils.ts [hunk 2]
```diff
@@ -30,3 +31,33 @@ export function filesToWebContainerFs(
...
+        const node = current[dir]
+        if (!('directory' in node))
+          throw new Error('Unexpected directory but found file')
+        current = node.directory
+      }
+      current[filename] = { file }
+    }
+  }
+  return tree
+}
```
Comment (serkodev) [inline templates/utils.ts:63]: I think the code here (`filesToVirtualFsTree` function) is somewhat repetitive above, but I'm not sure whether it should be merged with the functionality above, or if the whole approach should be redesigned.

**Review by antfu — APPROVED** [review]

**Comment by antfu** [comment]
That's a good start, thank you!

File: templates/utils.ts (hunk 2 above)
Comment (antfu) [reply templates/utils.ts:63]: I am ok with keeping them as is, as the logic is basically creating trees and we do need to over-refactor our code before we need that.

**Pull request merged**